provisioning/device/samples/register_symkey.js

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var iotHubTransport = require('azure-iot-device-mqtt').Mqtt;
+var Client = require('azure-iot-device').Client;
+var Message = require('azure-iot-device').Message;
+
+var ProvisioningTransport = require('azure-iot-provisioning-device-http').Http;
+// Feel free to change the preceding using statement to anyone of the following if you would like to try another protocol.
+// var ProvisioningTransport = require('azure-iot-provisioning-device-amqp').Amqp;
+// var ProvisioningTransport = require('azure-iot-provisioning-device-amqp').AmqpWs;
+// var ProvisioningTransport = require('azure-iot-provisioning-device-mqtt').Mqtt;
+// var ProvisioningTransport = require('azure-iot-provisioning-device-mqtt').MqttWs;
+
+var symmetricKeySecurity = require('azure-iot-security-symmetric-key');
+var ProvisioningDeviceClient = require('azure-iot-provisioning-device').ProvisioningDeviceClient;
+
+//
+// For the public clouds the address of the provisioning host would be: global.azure-devices-provisioning.net
+//
+var provisioningHost = '<the host for provisioning>';
+
+//
+// You can find your idScope in the portal overview section for your dps instance.
+//
+var idScope = '<id scope for dps instance>';
+
+//
+// The registration id of the device to be registered.
+//
+var registrationId = '<registration id for device>';
+
+var symmetricKey = '<symmetric key for device>';
+
+var provisioningSecurityClient = new symmetricKeySecurity.SymmetricKeySecurityClient(symmetricKey, registrationId);
```

Comment:
>symmetricKey, registrationId [](start = 85, length = 28)

nit: for some reason, i would find it more natural if these two were reverted (registrationId first). I wonder if there are other APIs in the SDK that would suggest this too or not.

Reply:
Fine.

---
In reply to: [231345275](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231345275) [](ancestors = 231345275)